src/server.cpp

```diff
@@ -0,0 +1,177 @@
+
+bool
+server_t::initialize(const rapidjson::Value& config) {
+    if (!config.HasMember("locators")) {
+        std::cerr << "'locators' field is missed in config. You should specify locators as a list of strings in format 'host:port'." << std::endl;
+        return false;
+    }
+
+    auto& locators = config["locators"];
+
+    if (!locators.IsArray()) {
+        std::cerr << "Locators must be specified as an array of endpoints." << std::endl;
+        return false;
+    }
+
+    std::vector<framework::service_manager_t::endpoint_t> endpoints;
+
+    for (size_t i = 0; i < locators.Size(); ++i) {
+        if (!locators[i].IsString()) {
+            std::cerr << "Bad format of locator's endpoint. It must be a string in format 'host:port'." << std::endl;
+            return false;
+        }
+
+        std::string locator = locators[i].GetString();
+
+        // kostyl-way!
+        size_t delim = locator.rfind(':');
+        if (delim == std::string::npos) {
+            std::cerr << "Bad format of locator's endpoint. Format: host:port." << std::endl;
+            return false;
+        }
+
+        std::string host = locator.substr(0, delim);
+        std::uint16_t port;
+        std::istringstream port_parser(locator.substr(delim + 1));
+        if (!(port_parser >> port)) {
+            std::cerr << "Bad format of locator's endpoint. Format: host:port." << std::endl;
+            return false;
+        }
+
+        endpoints.emplace_back(host, port);
+    }
+
+    if (endpoints.empty()) {
+        std::cerr << "List of locators is empty. Specify some ones." << std::endl;
+        return false;
+    }
+
+    std::string prefix = "native-proxy";
+    if (config.HasMember("logging_prefix") && config["logging_prefix"].IsString()) {
+        prefix = config["logging_prefix"].GetString();
+    }
+    CP_DEBUG("successfully parsed logging prefix field: %s", prefix);
+
+    std::size_t threads = threads_count();
+    if (config.HasMember("threads")) {
+        if (config["threads"].IsUint() && config["threads"].GetUint() > 0) {
+            threads = config["threads"].GetUint();
+        } else {
+            CP_ERROR("failed to parse config: thread number field must be a positive integer");
+            return false;
+        }
+    }
+    CP_DEBUG("successfully parsed thread number field: %d", threads);
+
+    service_manager = framework::service_manager_t::create(endpoints, prefix, threads);
+
+    pool_size = 10;
+    if (config.HasMember("service_pool")) {
+        pool_size = config["service_pool"].GetUint();
+    }
+
+    reconnect_timeout = 180;
+    if (config.HasMember("reconnect_timeout")) {
+        reconnect_timeout = config["reconnect_timeout"].GetUint();
+    }
+
+    request_timeout = 5;
+    if (config.HasMember("request_timeout")) {
+        request_timeout = config["request_timeout"].GetUint();
+    }
+
+    CP_INFO("proxy has successfully started");
+
+    on<handler::ping_t>   (options::prefix_match("/ping"));
```

Comment:
I don't know how to implement this functionality using thevoid without splitting headers and path handlers.